docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/RegressionTemplate.txt

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Data;
+
+namespace Microsoft.ML.Samples.Dynamic.Trainers.Regression
```

Comment:
Since this is a txt, do we get Intellisence in VS?

Reply:
Its is a stub code (an agreed upon template) e.g. I took it from Shahab's samples for regression and made it as template. You wont want get Intellisence for it but you get Intellisence for the generated file like `FastTreeTemplate.cs`.

---
In reply to: [267094827](https://github.com/dotnet/machinelearning/pull/3017#discussion_r267094827) [](ancestors = 267094827)

Reply:
I'm wondering if managing lots of txt files will be more work than standardizing files directly. I'm on the fence.

---
In reply to: [267099333](https://github.com/dotnet/machinelearning/pull/3017#discussion_r267099333) [](ancestors = 267099333,267094827)

Reply:
I think there will be one file for each ML task we have e.g. Regression, Classification etc. and that also in case when we have multiple samples using same pattern and one .tt for each sample. The .cs file will be auto generated.

So, this is how it will work. Create a sample in a normal way using .cs file. Once the code is agreed upon and there are a couple of places the same pattern is used then turn that into .tt file and make placeholders in it. Also, if there is a problem in template, the generated csharp file will have the same problem as the template file. Any indentation errors etc. will definitely be caught in the review (or even during development). 

I think t4 template is just a way of automatically doing copy-paste.


---
In reply to: [267100884](https://github.com/dotnet/machinelearning/pull/3017#discussion_r267100884) [](ancestors = 267100884,267099333,267094827)

Reply:
i think in the long term having templates like this will improve maintainability and consistency; also to me it's easier to to than copy paste.

---
In reply to: [267112240](https://github.com/dotnet/machinelearning/pull/3017#discussion_r267112240) [](ancestors = 267112240,267100884,267099333,267094827)

Reply:
also I think we can get away with having to hard code the outputs and have the template calculate and print it directly. that would simplify the .tt files substantially. I'm working on that in my new PR.

---
In reply to: [267114254](https://github.com/dotnet/machinelearning/pull/3017#discussion_r267114254) [](ancestors = 267114254,267112240,267100884,267099333,267094827)

Reply:
In corefx/coreclr we using the file extension `.ttinclude`.

https://github.com/dotnet/corefx/blob/a0e5f73d0c0876840bc4f3745e1ca8385a91cfcf/src/System.Private.Xml/src/System/Xml/Core/XmlEncodedRawTextWriter.tt#L5

https://github.com/dotnet/corefx/blob/5f2e78f0bd7d2bee1b6e332edc176e87283bdaed/src/Common/src/CoreLib/System/Numerics/ConstantHelper.tt#L15

https://github.com/dotnet/corefx/blob/475164edf559787c016676e561257446594ce5a0/src/System.Numerics.Tensors/tests/TensorOperations.tt#L4